packages/clay-autocomplete/src/ClayAutocomplete.js

```diff
@@ -242,19 +253,32 @@ ClayAutocomplete.STATE = {
 	data: Config.object(),
 
 	/**
-	 * The array of data items that the data source contains or
-	 * the URL for the data provider to request.
-	 * @default undefined
+	 * The array of data items that the data source contains,
+	 * the URL for the data provider to request, or a function
+	 * that receives the query and returns a promise with the
+	 * elements.
+	 *
 	 * @instance
+	 * @default undefined
 	 * @memberof ClayAutocomplete
-	 * @type {!(string|object|array)}
+	 * @type {!(string|object|array|function)}
 	 */
 	dataSource: Config.oneOfType([
-		Config.string(),
-		Config.object(),
 		Config.array(),
+		Config.func(),
+		Config.object(),
+		Config.string(),
 	]).required(),
 
+	/**
+	 * Set the request debounce time
+	 * @instance
+	 * @default 200
+	 * @memberof ClayDataProvider
```

Comment:
Just certified to go according to the other standards we have. `@memberof ClayAutocomplete`